src/Type/ObjectType/MenuItem.php

```diff
@@ -21,7 +21,7 @@ public static function register_type() {
 			'MenuItem',
 			[
 				'description' => __( 'Navigation menu items are the individual items assigned to a menu. These are rendered as the links in a navigation menu.', 'wp-graphql' ),
-				'interfaces'  => [ 'Node', 'DatabaseIdentifier' ],
+				'interfaces'  => [ 'Node', 'DatabaseIdentifier', 'UniformResourceIdentifiable' ],
```

Comment:
Ah, so even though the `MenuItem` type has a `uri` field, Menu Items are not actually uniquely identifiable by a URI. 

If you take the `MenuItem.uri` field and do a `nodeByUri( $uri: MenuItem.uri ) { ... }` query, you will get another node (Post, Page, etc)

The `MenuItem` Type should not implement the `UniformResourceIdentifiable` interface, as a MenuItem is not identifiable by a URI. It links to other nodes that are identifiable by uri, but it by itself is not.

Reply:
If we look in the Schema now for the `nodeByUri` field, we see that it returns `UniformResourceIdentifiable` Interface, which now has `MenuItem` as a Possible Type, which is misleading, as `nodeByUri` will never return an object with type `MenuItem`

![CleanShot 2022-05-13 at 16 32 49](https://user-images.githubusercontent.com/1260765/168397401-689ed35f-a530-494f-9c05-32a84d6523e6.png)